src/Metadata/Property/PropertyMetadata.php

```diff
@@ -60,7 +60,7 @@ public function getType()
      *
      * @param Type $type
      *
-     * @return self
+     * @return PropertyMetadata
```

Comment:
:-1: to change this. `self` is perfectly valid according to the current and future format and ease the refactoring.